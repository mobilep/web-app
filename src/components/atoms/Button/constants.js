export const COLORS = {
	DEFAULT: false,
	GREY: 'grey',
	RED: 'red',
	WHITE: 'white',
};
