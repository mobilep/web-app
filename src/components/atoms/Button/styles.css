:root {
	--Button-transition: .1s ease-in-out;
}

.Button {
	background-color: var(--tomatoRed);
	border-radius: var(--borderRadius);
	border-width: 1px;
	border-style: solid;
	border-color: var(--tomatoRed);
	color: var(--white);
	cursor: pointer;
	font-size: 14px;
	padding: 5px 16px;
	transition:
		background-color var(--Button-transition),
		border-color var(--Button-transition),
		opacity var(--Button-transition);
	user-select: none;
}

.Button-primary {
	font-size: 16px;
	min-width: 150px;
}

.Button:disabled {
	cursor: not-allowed;
	opacity: .7;
}

.Button-grey,
.Button-white {
	color: var(--tomatoRed);
}

.Button-grey {
	background-color: var(--grey1);
	border-color: var(--grey1);
}

.Button-white,
.Button-white:hover {
	background-color: var(--white);
	border-color: var(--white);
	color: var(--tomatoRed);
}

.Button-text,
.Button-text:hover {
	font-size: 16px;
	background-color: transparent;
	border: none;
	color: var(--tomatoRed);
}

@media (hover: hover) {
	.Button:hover {
		background-color: var(--darkRed);
		border-color: var(--darkRed);
		color: var(--white);
		text-decoration: none;
	}

	.Button-grey:hover {
		background-color: var(--white);
		border-color: var(--tomatoRed);
		color: var(--tomatoRed);
	}

	.Button-text:hover {
		background-color: transparent;
		text-decoration: underline;
		color: var(--darkRed);
	}
}
