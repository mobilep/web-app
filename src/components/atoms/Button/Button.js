import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { COLORS } from './constants';

import './styles.css';

const getClassName = ({ color, primary, className }) => classNames({
	Button: true,
	'Button-grey': color === COLORS.GREY,
	'Button-white': color === COLORS.WHITE,
	'Button-primary': primary,
}, className);

const Button = ({
	className = '',
	color = COLORS.DEFAULT,
	disabled = false,
	onClick = () => {},
	primary = false,
	children = '',
	type = 'button',
	activeClassName = '',
	to = '',
	isActive,
}) => {
	return to
		? (
			<NavLink
				className={getClassName({ color, primary, className })}
				activeClassName={activeClassName}
				to={to}
				isActive={isActive}
			>
				{children}
			</NavLink>
		)
		: (
			<button
				disabled={disabled}
				className={getClassName({ color, primary, className })}
				onClick={onClick}
				type={type}
			>
				{children}
			</button>
		);
};

Button.propTypes = {
	activeClassName: PropTypes.string,
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.element]),
	className: PropTypes.string,
	color: PropTypes.oneOf(Object.values(COLORS)),
	disabled: PropTypes.bool,
	isActive: PropTypes.func,
	onClick: PropTypes.func,
	primary: PropTypes.bool,
	to: PropTypes.string,
	type: PropTypes.string,
};

export default Button;
