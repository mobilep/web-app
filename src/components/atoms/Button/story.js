import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, selectV2 } from '@storybook/addon-knobs/react';
import { titleCaseKeys } from '../../../utils';

import Button from './Button';
import { COLORS } from './constants';

const CASED_COLORS = titleCaseKeys(COLORS);

const props = ({
	defaultText = 'Example',
} = {}) => ({
	children: text('Text', defaultText),
	color: selectV2('Color', CASED_COLORS),
	primary: boolean('Primary', true),
	disabled: boolean('Disabled', false),
	onClick: action('Clicked'),
});

storiesOf('1. Atoms / Button', module)
	.add('with text', () => <Button {...props()} />)
	.add('with emoji', () => <Button {...props({ defaultText: 'ðŸ˜Ž' })} />);
