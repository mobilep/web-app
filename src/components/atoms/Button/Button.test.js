import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('Atom - Button', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<Button>
				Text
			</Button>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render anchor tag', () => {
		const wrapper = shallow(
			<Button to='href'>
				Text
			</Button>
		);
		expect(wrapper).toMatchSnapshot();
	});

});

