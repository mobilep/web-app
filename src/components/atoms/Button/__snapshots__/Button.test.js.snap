// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - Button Should render 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Text
</button>
`;

exports[`Atom - Button Should render anchor tag 1`] = `
<NavLink
  activeClassName=""
  aria-current="page"
  className="Button"
  to="href"
>
  Text
</NavLink>
`;
